pkg/middlewares/metrics/metrics.go

```diff
@@ -74,13 +79,22 @@ func WrapServiceHandler(ctx context.Context, registry metrics.Registry, serviceN
 }
 
 func (m *metricsMiddleware) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	var labels []string
+	labels := make([]string, 0, len(m.baseLabels)+6)
```

Comment:
could you remove the `make`.

Reply:
I would prefer to keep it. We know the cap of the slice so it is optimized to initialize it with the correct cap so that following `append()` calls do not need to grow it.

Reply:
I would prefer to remove it

Reply:
These changes are required to merge the PR.

Reply:
If these changes are not made quickly, the PR will not be part of the next release.

Reply:
@ldez out of curiosity, what is your preference based on?

Reply:
- avoid magic number
- optimization need to be proved

![rules-optimization](https://phauer.com/blog/2020/02-wall-coding-wisdoms-quotes/rules-optimization.png)

Reply:
ok, no meaningful reason, thanks and have a nice day!

Reply:
There are no meaningful reason do use the `make` without knowing if this has an impact.

Reply:
```go
package main

import "testing"

func BenchmarkNoCap(b *testing.B) {
	for i := 0; i < b.N; i++ {
		var slice []string
		slice = append(slice, []string{"a", "q", "w"}...)
		slice = append(slice, []string{"z", "s", "x"}...)
	}
}

func BenchmarkCap(b *testing.B) {
	for i := 0; i < b.N; i++ {
		slice := make([]string, 0, 6)
		slice = append(slice, []string{"a", "q", "w"}...)
		slice = append(slice, []string{"z", "s", "x"}...)
	}
}
```

```
$ go test -bench . 
goos: darwin
goarch: amd64
BenchmarkNoCap-12    	 7565632	       146 ns/op
BenchmarkCap-12      	57868803	        21.0 ns/op
PASS
ok  	_/Users/s.rabot/tmp/go-cert	3.505s
```

Reply:
micro benchmarck prove nothing on real and concrete impact in the context of this code..